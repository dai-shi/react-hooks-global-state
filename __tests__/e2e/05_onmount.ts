import puppeteer from 'puppeteer';

jest.setTimeout(15 * 1000);

describe('05_onmount', () => {
  const port = process.env.PORT || '8080';

  it('should work with events', async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector('body > div > div > span');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await browser.close();
  });
});
