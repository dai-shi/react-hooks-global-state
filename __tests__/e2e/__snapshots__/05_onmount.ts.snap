// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`05_onmount should work with events 1`] = `
"
    <div id=\\"app\\"><div><h1>PageInfo</h1><span>sunt aut facere repellat provident occaecati excepturi optio reprehenderit</span></div><div><button type=\\"button\\">Random</button></div><div style=\\"color: red;\\"></div></div>
  

"
`;
